### ISSUE TITLE
ActiveRecord::RecordNotUnique on table "bigbluebutton_meetings"

### ISSUE THREAD

**Issue by mconf-daileon** [mconf/mconf-web#721]
---

Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Redmine Issue: 1536, http://dev.mconf.org/redmine/issues/1536

---

It's happening with different rooms. Was seen around 5 times in the past ~2 months.

```
An ActiveRecord::RecordNotUnique occurred in background at 2015-03-30 21:56:21 +0200 :

  Mysql2::Error: Duplicate entry '990a5cd0-f685-430c-942e-b021ec78f3bc-1397501921-2015-03-30 19:54' for key 'index_bigbluebutton_meetings_on_meetingid_and_start_time': INSERT INTO `bigbluebutton_meetings` (`created_at`, `creator_id`, `creator_name`, `meetingid`, `name`, `room_id`, `running`, `server_id`, `start_time`, `updated_at`) VALUES ('2015-03-30 19:56:20', 3, 'Felipe Cecagno', '990a5cd0-f685-430c-942e-b021ec78f3bc-1397501921', 'Felipe Cecagno', 3, 1, 1, '2015-03-30 19:54:06', '2015-03-30 19:56:20')
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:303:in `query'

  -------------------------------
Backtrace:
-------------------------------

  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:303:in `query'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:303:in `block in execute'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_adapter.rb:373:in `block in log'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_adapter.rb:367:in `log'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:303:in `execute'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/mysql2_adapter.rb:228:in `execute'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/mysql2_adapter.rb:250:in `exec_insert'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract/database_statements.rb:95:in `insert'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract/query_cache.rb:14:in `insert'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/relation.rb:64:in `insert'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/persistence.rb:502:in `_create_record'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/attribute_methods/dirty.rb:87:in `_create_record'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/callbacks.rb:306:in `block in _create_record'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:113:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:113:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:215:in `block in halting_and_conditional'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:215:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:215:in `block in halting_and_conditional'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:215:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:215:in `block in halting_and_conditional'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:86:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:86:in `run_callbacks'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/callbacks.rb:306:in `_create_record'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/timestamp.rb:57:in `_create_record'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/persistence.rb:482:in `create_or_update'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/callbacks.rb:302:in `block in create_or_update'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:113:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:113:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:166:in `block in halting'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:166:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:166:in `block in halting'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:166:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:166:in `block in halting'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:86:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activesupport-4.1.4/lib/active_support/callbacks.rb:86:in `run_callbacks'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/callbacks.rb:302:in `create_or_update'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/persistence.rb:103:in `save'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/validations.rb:51:in `save'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/attribute_methods/dirty.rb:21:in `save'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/transactions.rb:268:in `block (2 levels) in save'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/transactions.rb:329:in `block in with_transaction_returning_status'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract/database_statements.rb:201:in `block in transaction'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract/database_statements.rb:209:in `within_new_transaction'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract/database_statements.rb:201:in `transaction'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/transactions.rb:208:in `transaction'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/transactions.rb:326:in `with_transaction_returning_status'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/transactions.rb:268:in `block in save'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/transactions.rb:283:in `rollback_active_record_state!'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/transactions.rb:267:in `save'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/activerecord-4.1.4/lib/active_record/persistence.rb:34:in `create'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/bundler/gems/bigbluebutton_rails-62e490fd7dd7/app/models/bigbluebutton_room.rb:323:in `update_current_meeting'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/bundler/gems/bigbluebutton_rails-62e490fd7dd7/app/models/bigbluebutton_room.rb:126:in `fetch_meeting_info'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/bundler/gems/bigbluebutton_rails-62e490fd7dd7/app/workers/bigbluebutton_meeting_updater.rb:16:in `perform'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/resque-1.25.2/lib/resque/job.rb:227:in `perform'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/resque-1.25.2/lib/resque/worker.rb:250:in `perform'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/resque-1.25.2/lib/resque/worker.rb:189:in `block in work'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/resque-1.25.2/lib/resque/worker.rb:166:in `loop'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/resque-1.25.2/lib/resque/worker.rb:166:in `work'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/resque-1.25.2/lib/resque/tasks.rb:41:in `block (2 levels) in <top (required)>'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/task.rb:240:in `call'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/task.rb:240:in `block in execute'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/task.rb:235:in `each'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/task.rb:235:in `execute'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
  /home/mconf/.rbenv/versions/2.2.0/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/task.rb:165:in `invoke'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:150:in `invoke_task'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `each'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `block in top_level'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:115:in `run_with_threads'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:100:in `top_level'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:78:in `block in run'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:176:in `standard_exception_handling'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/lib/rake/application.rb:75:in `run'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/gems/rake-10.3.2/bin/rake:33:in `<top (required)>'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/bin/rake:23:in `load'
  /var/www/mconf-web/shared/bundle/ruby/2.2.0/bin/rake:23:in `<main>'

-------------------------------
Data:
-------------------------------

  * data: {:resque=>
    {:failed_at=>"2015-03-30 16:56:20 -0300",
     :queue=>"bigbluebutton_rails",
     :worker=>"hydrus:18885:*",
     :payload=>{"class"=>"BigbluebuttonMeetingUpdater", "args"=>[3]},
     :error_class=>"ActiveRecord::RecordNotUnique",
     :error_message=>
      "Mysql2::Error: Duplicate entry '990a5cd0-f685-430c-942e-b021ec78f3bc-1397501921-2015-03-30 19:54' for key 'index_bigbluebutton_meetings_on_meetingid_and_start_time': INSERT INTO `bigbluebutton_meetings` (`created_at`, `creator_id`, `creator_name`, `meetingid`, `name`, `room_id`, `running`, `server_id`, `start_time`, `updated_at`) VALUES ('2015-03-30 19:56:20', 3, 'Felipe Cecagno', '990a5cd0-f685-430c-942e-b021ec78f3bc-1397501921', 'Felipe Cecagno', 3, 1, 1, '2015-03-30 19:54:06', '2015-03-30 19:56:20')"}}

```

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Lucas Zawacki** (Lucas Zawacki)
Original Date: 2015-04-23T20:23:45Z

---

It seems the index uses the datetime up to the minutes which results in non unique indexes. This can easily be replicated by starting and stopping a conference quickly a few times.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2015-12-03T15:59:02Z

---

Still happening in the latest version.

**Comment by daronco** [comment]
This was probably solved by the PR mentioned above. The index that caused the issue was related to the attribute `start_time`, that was a date. Now it uses `create_time` instead, that is a timestamp and might not ever be duplicated.

I'll leave this in feedback until we can test it better.

**Comment by daronco** [comment]
I'm marking this as closed since it hasn't happened anymore. It was already solved on version 2.2.0, so the milestone and changelog should be fixed.
